packages/prosemirror-lwdita-demo/cypress/e2e/github-inegration.cy.ts

```diff
@@ -0,0 +1,307 @@
+    const mockCode = 'mock-user-code';
+    // Intercept the GitHub OAuth URL
+    const githubOAuthUrl = /https:\/\/github\.com\/login\/oauth\/authorize\?.*/;
+
+    // Intercept the OAuth request and mock the authentication process
+    cy.intercept('GET', githubOAuthUrl, {
+      statusCode: 302,
+      headers: { location: `http://localhost:1234/?code=${mockCode}&state=eyJnaHJlcG8iOiJldm9sdmVkYmluYXJ5L3Byb3NlbWlycm9yLWx3ZGl0YSIsInNvdXJjZSI6InBhY2thZ2VzL3Byb3NlbWlycm9yLWx3ZGl0YS1kZW1vL2V4YW1wbGUteGRpdGEvMDItc2hvcnQtZmlsZS54bWwiLCJicmFuY2giOiJtYWluIiwicmVmZXJlciI6Imh0d3NyaHRzaHJ0cyJ9` }
+    }).as('githubOAuth');
+
+    // Intercept the token request
+    const tokenRequest = /http:\/\/localhost:3000\/api\/github\/token\?.*/;
+    cy.intercept('GET', tokenRequest, {
+      statusCode: 200,
+      headers: { 'content-type': 'application/json' },
+      body: {
+        token: 'mock-token'
+      }
+    }).as('requestToken');
+
+    cy.visit('http://localhost:1234/?ghrepo=evolvedbinary/prosemirror-lwdita&source=packages/prosemirror-lwdita-demo/example-xdita/02-short-file.xml&branch=main&referer=https://petal.evolvedbinary.com/');
+    cy.get('#publishFile').should('exist');
+  });
+});
+
+describe('PR dialog', () => {
+
+  beforeEach(() => {
+    // mock the GitHub OAuth URL
+    const mockCode = 'mock-user-code';
+    // Intercept the GitHub OAuth URL
+    const githubOAuthUrl = /https:\/\/github\.com\/login\/oauth\/authorize\?.*/;
+
+    // Intercept the OAuth request and mock the authentication process
+    cy.intercept('GET', githubOAuthUrl, {
+      statusCode: 302,
+      headers: { location: `http://localhost:1234/?code=${mockCode}&state=eyJnaHJlcG8iOiJldm9sdmVkYmluYXJ5L3Byb3NlbWlycm9yLWx3ZGl0YSIsInNvdXJjZSI6InBhY2thZ2VzL3Byb3NlbWlycm9yLWx3ZGl0YS1kZW1vL2V4YW1wbGUteGRpdGEvMDItc2hvcnQtZmlsZS54bWwiLCJicmFuY2giOiJtYWluIiwicmVmZXJlciI6Imh0d3NyaHRzaHJ0cyJ9` }
+    }).as('githubOAuth');
+
+    // Intercept the token request
+    const tokenRequest = /http:\/\/localhost:3000\/api\/github\/token\?.*/;
+    cy.intercept('GET', tokenRequest, {
+      statusCode: 200,
+      headers: { 'content-type': 'application/json' },
+      body: {
+        token: 'mock-token'
+      }
+    }).as('requestToken');
+
+    cy.visit('http://localhost:1234/?ghrepo=evolvedbinary/prosemirror-lwdita&source=packages/prosemirror-lwdita-demo/example-xdita/02-short-file.xml&branch=main&referer=https://petal.evolvedbinary.com/');
+  
+    cy.get('body > div.toastify.on.toast.toast__panel.toast--welcome.toastify-right.toastify-top > section > button').click();
+  })
+
+  it('should open the PR dialog', () => {
+    cy.get('#publishFile').click();
+    cy.get('#prDialog').should('exist');
+  });
+
+  it('should close the PR dialog', () => {
+    cy.get('#publishFile').click();
+    cy.get('#closeButton').click();
+    cy.get('#prDialog').should('not.exist');
+  });
+
+  it.skip('should close the PR dialog when clicking outside the dialog', () => {
```

Comment:
We should add a comment about why we are skipping this at the moment, and also add a card to our backlog to address this please.

Reply:
Fixed in c8cbaab0c1c7ff2a5e9ce137cf7988844318f435